### PR TITLE
Chrome Manifest v3 Support

### PR DESCRIPTION
**Pull request by BenMcGarry**
RES Manifest v3 Support

https://developer.chrome.com/docs/extensions/mv3/intro/mv3-migration/

Firefox will remain MV2.

### CODE REVIEW

**Comment by BenMcGarry** [comment]
Made some progress, service workers now working. New errors.

`TypeError: Error in invocation of scripting.executeScript(scripting.ScriptInjection injection, optional function callback): No matching signature.` All over, this fires any time we try to run any script...


`Uncaught (in promise) Error: The message port closed before a response was received. `https://github.com/honestbleeps/Reddit-Enhancement-Suite/blob/master/lib/environment/utils/api.js#L8

File: lib/environment/background/loadScript.js [hunk 1]
```diff
@@ -4,5 +4,5 @@ import { apiToPromise } from '../utils/api';
 import { addListener } from './messaging';
 
 addListener('loadScript', async ({ url }, { tab: { id: tabId }, frameId }) => {
-	await apiToPromise(chrome.tabs.executeScript)(tabId, { file: url, frameId, runAt: 'document_start' });
+	await apiToPromise(chrome.scripting.executeScript)(tabId, { file: url, frameId, runAt: 'document_start' });
```
Comment (larsjohnsen) [inline lib/environment/background/loadScript.js:7]: The `executeScript` function [has changed a little](https://developer.chrome.com/docs/extensions/reference/scripting/#injection-targets):
```
addListener('loadScript', async ({ url }, { tab: { id: tabId }, frameId }) => {
	await apiToPromise(chrome.scripting.executeScript)({
		target: { tabId, frameIds: [frameId] },
		files: [url],
	});
});
```

Comment (BenMcGarry) [reply lib/environment/background/loadScript.js:7]: This solves the script error, thank you!

**Comment by BenMcGarry** [comment]
Chrome compat appears to be good. Firefox is hard to test as their implementation is not yet complete. We could split the manifest now and push v3 early however i'll wait to test FF some more.

**Comment by BenMcGarry** [comment]
@larsjohnsen Have you played with MV3 at all? With the latest changes Chrome is fine, but Firefox appears to be hanging up on some CSP issues I think?

**Comment by BenMcGarry** [comment]
Known Issues:

`Uncaught (in promise) Error: The message port closed before a response was received. `https://github.com/honestbleeps/Reddit-Enhancement-Suite/blob/master/lib/environment/utils/api.js#L8

**Comment by BenMcGarry** [comment]
MV3 calls appear to be good for Chrome, testing in FF appears stable on MV2 with MV3 compat.

**Pull request merged**